src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6308,71 +6308,131 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
+        noway_assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        // As we output multiple instructions for SIMD zeroing, we want to balance code size, with throughput
+        // so cap max size at 16 * MAX SIMD length
+        int initMaxSIMDSize = 16 * XMM_REGSIZE_BYTES;
+        if ((untrLclHi - untrLclLo) <= initMaxSIMDSize)
+        {
+            /*
```

Comment:
Should https://github.com/dotnet/runtime/blob/8079a27a050cb0bf7da5851ef454a34423dd0715/src/coreclr/src/jit/codegencommon.cpp#L4743 need to be updated to match? We may be doing unnecessary spills now.

Reply:
Should be doing the same as the `rep stosd` branch? (which is the else of this, indented in PR) 

Although the `rep stosd` branch has much more register shuffles as it needs 3 to call `rep stosd` whereas the simd branch doesn't need any that could be in use.

e.g. in the above example 
```asm
-   mov      ecx, 60
-   xor      rax, rax
-   rep stosd          ; 60 * 4 = 240 bytes
```
Is the same as
```diff
+   movdqu   xmmword ptr [rsp+40H], xmm4
...
+   movdqu   xmmword ptr [rsp+120H], xmm4
```
288 (120H) - 64 (40H) +16 = 240 bytes

Reply:
Noticed that was clearly larger than 128 bytes so updated title 😄